src/store/channels/saga.ts

```diff
@@ -81,6 +81,8 @@ export function* openConversation(conversationId) {
   }
 
   yield call(reset);
+  // cancel start of create conversation flow if it's in progress (resolves the race condition)
+  yield put({ type: CreateConversationSagaActionTypes.Cancel });
```

Comment:
This cancels the "startConversation", which resolves the race condition.

<img width="590" alt="image" src="https://github.com/zer0-os/zOS/assets/33264364/06976863-fb39-4778-bed1-950aaf1ee71d">

Not sure if this is the correct place, since this will also be called whenever user just clicks on a conversation from the conversation list panel. But the otherway would be to call it from the frontend i guess (from `createOneOnOneConversation` in `messenger/list/index` file..)

Reply:
Yeah, I think the problem is that we ended up putting business logic in the component (https://github.com/zer0-os/zOS/blob/main/src/components/messenger/list/index.tsx#L143)

That should probably just call up to the sagas where we can handle bailing out of the flow properly rather than trying to determine that a conversation exists within the component.